environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentStackConfigUpdateService.java

```diff
@@ -0,0 +1,40 @@
+package com.sequenceiq.environment.environment.service;
+
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.flow.EnvironmentReactorFlowManager;
+import org.springframework.stereotype.Service;
+
+@Service
+public class EnvironmentStackConfigUpdateService {
+    private final EnvironmentService environmentService;
+
+    private final EnvironmentReactorFlowManager reactorFlowManager;
+
+    public EnvironmentStackConfigUpdateService(
+        EnvironmentService environmentService,
+        EnvironmentReactorFlowManager reactorFlowManager) {
+        this.environmentService = environmentService;
+        this.reactorFlowManager = reactorFlowManager;
+    }
+
+    public void updateAllStackConfigsByCrn(String crn) {
+        String accountId = Crn.fromString(crn).getAccountId();
+        EnvironmentDto environment = environmentService.getByCrnAndAccountId(crn, accountId);
```

Comment:
why are you fetching the env here if you fetch it in `triggerStackConfigUpdates` again? this method actually uses the same `findByResourceCrnAndAccountIdAndArchivedIsFalse`, just calls a converter later

Reply:
Yeah that can be cleaned up a little.